packages/package-requester/src/packageRequester.ts

```diff
@@ -256,7 +256,12 @@ async function resolveAndFetch (
       resolvedVia,
       updated,
     },
-    bundledManifest: fetchResult.bundledManifest,
+    bundledManifest: manifest
+      ? fetchResult.bundledManifest
+      : () => Promise.resolve({
+        name: wantedDependency.alias,
+        version: wantedDependency.pref,
+      } as DependencyManifest),
```

Comment:
the tests in `@pnpm/package-requester` fail because of this change. I don't think this package should be changed at all.